tubular/dates.py

```diff
@@ -663,3 +663,344 @@ def transform(self, X):
+                )
+
+        if np.isnan(hour):
+            return np.nan
+        else:
+            return self.timeofday_mapping[hour]
+
+    def identify_timeofmonth(self, day):
+        """Map timeofmonth from day
+
+        Parameters
+        ----------
+        day : numeric
+            Day to map
+
+        Returns
+        -------
+        str : str
+            Mapped day
+        """
+
+        if not type(day) is float:
+            if not type(day) is int:
+                raise TypeError(f"{self.classname()}: day should be float or int")
+
+        if not np.isnan(day):
+            if day not in np.arange(1, 32, 1):
+                raise ValueError(
+                    f"{self.classname()}: day should be a whole number in 1-31"
+                )
+
+        if np.isnan(day):
+            return np.nan
+        else:
+            return self.timeofmonth_mapping[day]
+
+    def identify_timeofyear(self, month):
+        """Map timeofyear from month
+
+        Parameters
+        ----------
+        month : numeric
+            Month to map
+
+        Returns
+        -------
+        str : str
+            Mapped month
+        """
+
+        if not type(month) is float:
+            if not type(month) is int:
+                raise TypeError(f"{self.classname()}: month should be float or int")
+
+        if not np.isnan(month):
+            if month not in np.arange(1, 13, 1):
+                raise ValueError(
+                    f"{self.classname()}: month should be a whole number in 1-12"
+                )
+
+        if np.isnan(month):
+            return np.nan
+        else:
+            return self.timeofyear_mapping[month]
+
+    def identify_dayofweek(self, day):
+        """Map dayofweek from day
+
+        Parameters
+        ----------
+        day : numeric
+            Day to map
+
+        Returns
+        -------
+        str : str
+            Mapped day
+        """
+
+        if not type(day) is float:
+            if not type(day) is int:
+                raise TypeError(f"{self.classname()}: day should be float or int")
+
+        if not np.isnan(day):
+            if day not in np.arange(0, 7, 1):
+                raise ValueError(f"{self.classname()}: day should be in 0-6")
+
+        if np.isnan(day):
+            return np.nan
+        else:
+            return self.dayofweek_mapping[day]
+
+    def transform(self, X):
+        """Transform - Extracts new features from datetime variables
+
+        Parameters
+        ----------
+        X : pd.DataFrame
+            Data with columns to extract info from.
+
+        Returns
+        -------
+        X : pd.DataFrame
+            Transformed input X with added columns of extracted information.
+        """
+
+        X = super().transform(X)
+
+        for col in self.columns:
+            if not X[col].dtype.name == "datetime64[ns]":
+                try:
+                    X[col] = X[col].dt.tz_localize(None)
+                except Exception:
```

Comment:
Can we get a more specific error than the base Exception class here?

Reply:
https://github.com/lvgig/tubular/pull/49/commits/d8400648522c97918daf76941b578e2a55b6e676